src/components/alerting.tsx

```diff
@@ -726,6 +726,13 @@ const AlertsDetailsPage_: React.FC<AlertsDetailsPageProps> = ({ history, match }
   // eslint-disable-next-line react-hooks/exhaustive-deps
   const labels: PrometheusLabels = React.useMemo(() => alert?.labels, [labelsMemoKey]);
 
+  const ruleLabelsMemoKey = JSON.stringify(rule?.labels);
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  const ruleLabels: PrometheusLabels = React.useMemo(() => rule?.labels, [ruleLabelsMemoKey]);
+
+  const combinedLabels = { ...labels, ...ruleLabels };
```

Comment:
This doesn't look right to me. If `labels` and `ruleLabels` contain the same key, this will take the value from `ruleLabels`, which seems wrong?

Reply:
This code has been removed in latest commit.